File: samples/app-complete-sample/csharp/template-bot-master-csharp/src/dialogs/examples/basic/MessagebackDialog.cs
using Microsoft.Bot.Builder.Dialogs;
using Microsoft.Teams.TemplateBotCSharp.Properties;
using System;
using System.Threading;
using System.Threading.Tasks;
namespace Microsoft.Teams.TemplateBotCSharp.Dialogs
{
    /// <summary>
    /// This is Message Back Dialog Class. Main purpose of this class is to show example of Message Back event
    /// </summary
    public class MessagebackDialog : ComponentDialog
    {
        public MessagebackDialog() : base(nameof(MessagebackDialog))
        {
            InitialDialogId = nameof(WaterfallDialog);
            AddDialog(new WaterfallDialog(nameof(WaterfallDialog), new WaterfallStep[]
            {
                BeginFormflowAsync,
            }));
        }

        private async Task<DialogTurnResult> BeginFormflowAsync(
WaterfallStepContext stepContext,
CancellationToken cancellationToken = default(CancellationToken))
        {
            if (stepContext == null)
            {
                throw new ArgumentNullException(nameof(stepContext));
            }

            //Set the Last Dialog in Conversation Data
            //stepContext.State.SetValue(Strings.LastDialogKey, Strings.LastDialogMessageBackDialog);
            await stepContext.Context.SendActivityAsync(Strings.MessageBackTitleMsg);

            return await stepContext.EndDialogAsync(null, cancellationToken);
        }
    }
}

File: samples/TeamsSDK/Archived/app-complete-sample/nodejs/server/dialogs/basic/messagebackDialog.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

const { WaterfallDialog, ComponentDialog } = require('botbuilder-dialogs');
const MESSAGEBACK = 'msgback';

/**
 * MessageBackDialog class extends ComponentDialog to handle message back interactions.
 */
class MessageBackDialog extends ComponentDialog {
    /**
     * Constructor for the MessageBackDialog class.
     * @param {string} id - The dialog ID.
     * @param {StatePropertyAccessor} conversationDataAccessor - The state property accessor for conversation data.
     */
    constructor(id, conversationDataAccessor) {
        super(id);
        this.conversationDataAccessor = conversationDataAccessor;
        // Define the conversation flow using a waterfall model.
        this.addDialog(new WaterfallDialog(MESSAGEBACK, [
            this.messageBackDialog.bind(this),
        ]));
    }

    /**
     * Begins the message back dialog.
     * @param {WaterfallStepContext} stepContext - The waterfall step context.
     * @returns {Promise<DialogTurnResult>} The result of the dialog turn.
     */
    async messageBackDialog(stepContext) {
        const currentState = await this.conversationDataAccessor.get(stepContext.context, {});
        currentState.lastDialogKey = "MessageBackReceiverDialog";
        await stepContext.context.sendActivity('This is Message Back example');
        return await stepContext.endDialog();
    }
}

exports.MessageBackDialog = MessageBackDialog;